### PR TITLE
feat: stargate adapter

### PR DESCRIPTION
**Pull request by kyle-layerzero**

### CODE REVIEW

File: adapters/stargate/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  blockTimestamp,
+}: BlockData): Promise<OutputSchemaRow[]> => {
+  const positions = await getUserBalancesAtBlock(blockNumber);
+
+  return positions.map((position) => ({
+    block_number: blockNumber,
+    timestamp: blockTimestamp,
+    user_address: position.user,
+    token_address: position.lpToken,
+    token_balance: BigInt(parseUnits(position.balance, 18)),
```
Comment (0xroll) [inline adapters/stargate/src/index.ts:42]: this should be returning underlying token asset and raw token balance

**Comment by 0xroll** [comment]
@kyle-layerzero thanks for the PR

please check the above, also the lp tokens has transfer events, is this accounted for?

user A LP -> transfer lp token to user B

please provide the subgraph gh repo as well

**Comment by kyle-layerzero** [comment]
@0xroll invited you to https://github.com/LayerZero-Labs-External/linea-balances, let me know if you need a public repo

**Comment by 0xroll** [comment]
@kyle-layerzero i think the transfer events also have to be listened to as well?

https://lineascan.build/address/0xaad094f6a75a14417d39f04e690fc216f080a41a#writeContract#F23

not seeing it here : 

image

**Comment by 0xroll** [comment]
this pr will only be tracking positions in farming contract.

@kyle-layerzero do make sure to share with your community that only staked positions will get points awarded!

@nitish-91 lets merge this

File: adapters/stargate/src/index.ts (hunk 1 above)
Comment (0xroll) [reply adapters/stargate/src/index.ts]: @kyle-layerzero 
the token_address should not be the lptoken. it should be the underlying token, and the balance should also be underlying token balance instead.

**Pull request merged**